contracts/rootchain/RootChain.sol

```diff
@@ -3,220 +3,315 @@ pragma solidity ^0.4.24;
+            created_at: block.timestamp,
+            state: 1
         });
 
-        currentDepositBlock = currentDepositBlock.add(1);
-        emit Deposit(txList[10].toAddress(), msg.value);
+        emit StartedDepositExit(nonce, owner, amount);
     }
 
-
-    /// @param txPos [0] Plasma block number in which the transaction occured
-    /// @param txPos [1] Transaction Index within the block
-    /// @param txPos [2] Output Index within the transaction (either 0 or 1)
-    /// @param sigs First 130 bytes are signature of transaction and the rest is confirm signature
-    /// @notice Each signature is 65 bytes
-    function startExit(uint256[3] txPos, bytes txBytes, bytes proof, bytes sigs)
+    // Transaction encoding:
+    // [Blknum0, TxIndex0, Oindex0, depositNonce0, Amount0, ConfirmSig0
+    //  Blknum1, TxIndex1, Oindex1, depositNonce1, Amount1, ConfirmSig1
+    //  NewOwner0, Denom0, NewOwner1, Denom1, Fee]
+    //
+    // @param txPos   location of the transaction [blkNum, txIndex, outputIndex]
+    // @param txBytes raw transaction bytes
+    // @param proof   merkle proof of inclusion in the child chain
+    // @param sigs    signatures of transaction including confirm signatures
+    // @param confirmSignatures confirm signatures sent by the owners of the inputs acknowledging the spend. 
+    // @notice `confirmSignatures` and `ConfirmSig0`/`ConfirmSig1` are unrelated to each other.
+    function startTransactionExit(uint256[3] txPos, bytes txBytes, bytes proof, bytes sigs, bytes confirmSignatures)
         public
         payable
-        returns (uint256)
     {
-        // txBytes verification
         RLPReader.RLPItem[] memory txList = txBytes.toRlpItem().toList();
-        require(txList.length == 15, "incorrect tx length");
-        require(msg.sender == txList[10 + 2 * txPos[2]].toAddress(), "address mismatch");
-        require(msg.value == minExitBond, "incorrect exit bond");
-
-        uint256 priority = 1000000000*txPos[0] + 10000*txPos[1] + txPos[2];
+        require(txList.length == 17, "incorrect tx length");
+        require(msg.sender == txList[12 + 2 * txPos[2]].toAddress(), "caller must own this utxo");
+        require(msg.value >= minExitBond, "insufficient exit bond");
+        if (msg.value > minExitBond) {
+            uint256 excess = msg.value.sub(minExitBond);
+            balances[msg.sender] = balances[msg.sender].add(excess);
+            totalWithdrawBalance = totalWithdrawBalance.add(excess);
+        }
 
-        // creating the correct merkle leaf
+        // check proof and signatures
         bytes32 txHash = keccak256(txBytes);
+        bytes32 merkleHash = keccak256(abi.encodePacked(txHash, sigs));
+        bytes32 confirmationHash = keccak256(abi.encodePacked(merkleHash, childChain[txPos[0]].root));
+        require(txHash.checkSigs(confirmationHash,
+                                 txList[0].toUint() > 0 || txList[3].toUint() > 0, // existence of input0. Either a deposit or utxo
```

Comment:
could we pass in boolean variables called `input0Exists` and `input1Exists` to checkSigs instead of writing `txList[0].toUint() > 0 || txList[3].toUint() > 0` and the inline comment

Reply:
I did this because of the `Stack too deep` error. The function has too many variables being pushed onto the stack. The refactor however got rid of one of those lines. We always assume by construction that `input1` is always true. The only thing we need to check is `input2`